src/stats/archive/calculator.rs

```diff
@@ -0,0 +1,126 @@
+use super::loader::ArchiverRecord;
+use chain_impl_mockchain::block::BlockDateParseError;
+use chain_time::TimeEra;
+use itertools::Itertools;
+use jormungandr_lib::interfaces::BlockDate;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+use thiserror::Error;
+
+pub struct ArchiveStats {
+    records: Vec<ArchiverRecord>,
+}
+
+impl From<Vec<ArchiverRecord>> for ArchiveStats {
+    fn from(records: Vec<ArchiverRecord>) -> Self {
+        Self { records }
+    }
+}
+
+impl ArchiveStats {
+    pub fn calculate_distribution<T>(
+        result: &BTreeMap<T, usize>,
+    ) -> Result<BTreeMap<usize, usize>, ArchiveCalculatorError> {
+        let mut distribution = BTreeMap::new();
+
+        for item in 0..*result
+            .values()
+            .max()
+            .ok_or(ArchiveCalculatorError::EmptyResult)?
+            + 1
+        {
+            distribution.insert(item, 0);
+        }
```

Comment:
it's better to use `distribution.entry(count).or_default() += 1` and remove this initialization